ThreadLite.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright © 2013 Ivan Seidel Gomes.
+// Copyright © 2018 Stanislav Hnatiuk.
+
+#include "ThreadLite.h"
+
+ThreadLite::ThreadLite(void (*callback)(void), TIME_INT _interval) {
+  _onRun = callback;
+  _cached_next_run = 0;
+  last_run = 0;
```

Comment:
Avoiding to initialize last_run may lead to missed first interval.
For example, if we set very small interval, for example 10ms and current millis() is greater than 32780 (we still use unsigned type, so it can be) then _cached_next_run will be 10 (lasr_run + interval) and sign bit will be set for all the operations until we overflow and thus should_run() will return false.
So for the first run we have to wait around 32 seconds instead of 10ms.

It's kind of rare situation, but I think last_run should be initialized to millis() here.